### PR TITLE
Odst 125 admin cc request an account so that i can monitor a specific squadron group wing

### PR DESCRIPTION
**Pull request by Mikehickey7**
Password has regex on it, however the rules are not clear yet, this will be solved with the implementation of the password-strength library

### CODE REVIEW

**Review by yurisim — CHANGES_REQUESTED** [review]
Please check that e2e tests pass

File: apps/ods-api/prisma-seed.ts [hunk 1]
```diff
@@ -9,6 +9,31 @@ const orgName = '552 ACNS';
 const CCEmail = 'john.doe@us.af.mil';
 
 async function main() {
+  const userCreateInputs: Prisma.UserCreateInput[] = [
```
Comment (yurisim) [inline apps/ods-api/prisma-seed.ts:12]: Why is this here?

Comment (REDRICK696) [reply apps/ods-api/prisma-seed.ts:12]: idk

File: apps/ods/src/app/login/login.component.html [hunk 2]
```diff
@@ -8,7 +8,9 @@
     >
       <div>Don't have an account?</div>
 
-      <p class="underline text-right">Get one!</p>
+      <p class="underline text-right" [routerLink]="['/request-account']">
```
Comment (yurisim) [inline apps/ods/src/app/login/login.component.html:11]: Use the primary color styling to make it obvious that this is a link

File: libs/shared/angular/src/lib/passwordValidator.service.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+}
+//this is the password requirements regex
+export const regExps: { [key: string]: RegExp } = {
+  password:
+    /^(?=.*[0-9])(?=.*[~!@$%^&])(?=.*?[A-Z])(?=.*?[a-z])[a-zA-Z0-9~!@#$%^&]{8,48}$/,
+};
+//these are the custom messages for errors states
+export const errorMessages: { [key: string]: string } = {
+  password:
+    'Password must be between 8 and 48 characters, and contain at least one uppercase letter, one lowercase letter, one number and one special character',
```
Comment (yurisim) [inline libs/shared/angular/src/lib/passwordValidator.service.ts:27]: Specify what allowed types of special characters in this string.

**Review by yurisim — CHANGES_REQUESTED** [review]
Fix requested changes please

File: libs/shared/angular/src/lib/passwordValidator.service.ts (hunk 3 above)
Comment (yurisim) [reply libs/shared/angular/src/lib/passwordValidator.service.ts:27]: image

File: apps/ods/src/app/request-account/request-account.component.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      grade: [''],
+      permissions: ['', [Validators.required]],
+      org: ['', [Validators.required]],
+      password: [
+        '',
+        [Validators.required, Validators.pattern(regExps['password'])],
+      ],
+      confirmPassword: ['', [Validators.required]],
+    },
+    { validator: CustomValidators.checkPasswords }
```
Comment (yurisim) [inline apps/ods/src/app/request-account/request-account.component.ts:36]: Fix the deprecated custom validator syntax.

https://blog.angular-university.io/angular-custom-validators/

**Review by yurisim — CHANGES_REQUESTED** [review]

File: apps/ods/src/app/request-account/request-account.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      declarations: [RequestAccountComponent],
+      imports: [
+        ReactiveFormsModule,
+        FormsModule,
+        ApolloTestingModule,
+        MatCardModule,
+        MatInputModule,
+        MatIconModule,
+        MatButtonModule,
+        MatOptionModule,
```
Comment (yurisim) [inline apps/ods/src/app/request-account/request-account.component.spec.ts]: Is this needed?

**Review by yurisim — APPROVED** [review]

**Pull request merged**